jina/executors/encoders/frameworks.py

```diff
@@ -54,16 +58,23 @@ def _append_outputs(input_fn, outputs_name_to_append, output_fn):
         onnx.save(model, output_fn)
 
 
-class BaseTFEncoder(BaseTFExecutor, BaseEncoder):
-    pass
+class BaseTFEncoder(BaseEncoder, BaseTFDeviceHandler):
+    @property
+    def run_on_gpu(self):
+        return self.on_gpu
+
 
+class BaseTorchEncoder(BaseEncoder, BaseTorchDeviceHandler):
 
-class BaseTorchEncoder(BaseTorchExecutor, BaseEncoder):
-    pass
+    @property
+    def run_on_gpu(self):
```

Comment:
This pretends to add as a `bridge` from executors to DeviceHandler, to avoid removing `on_gpu` from `Executors`. I think it is something that is already used, and can be part of the reconstruction of `executors`.